components/prism-php.js

```diff
@@ -1,118 +1,204 @@
 /**
  * Original by Aaron Harun: http://aahacreative.com/2012/07/31/php-syntax-highlighting-prism/
  * Modified by Miles Johnson: http://milesj.me
+ * Rewritten by Tom Pavelec
  *
- * Supports the following:
- * 		- Extends clike syntax
- * 		- Support for PHP 5.3+ (namespaces, traits, generators, etc)
- * 		- Smarter constant and function matching
+ * Supports PHP 5.3 - 7.4
  *
  * Adds the following new token classes:
  * 		constant, delimiter, variable, function, package
  */
 (function (Prism) {
-	Prism.languages.php = Prism.languages.extend('clike', {
-		'keyword': /\b(?:__halt_compiler|abstract|and|array|as|break|callable|case|catch|class|clone|const|continue|declare|default|die|do|echo|else|elseif|empty|enddeclare|endfor|endforeach|endif|endswitch|endwhile|eval|exit|extends|final|finally|for|foreach|function|global|goto|if|implements|include|include_once|instanceof|insteadof|interface|isset|list|namespace|new|or|parent|print|private|protected|public|require|require_once|return|static|switch|throw|trait|try|unset|use|var|while|xor|yield)\b/i,
-		'boolean': {
-			pattern: /\b(?:false|true)\b/i,
-			alias: 'constant'
-		},
-		'constant': [
-			/\b[A-Z_][A-Z0-9_]*\b/,
-			/\b(?:null)\b/i,
-		],
-		'comment': {
-			pattern: /(^|[^\\])(?:\/\*[\s\S]*?\*\/|\/\/.*)/,
-			lookbehind: true
-		}
-	});
-
-	Prism.languages.insertBefore('php', 'string', {
-		'shell-comment': {
-			pattern: /(^|[^\\])#.*/,
-			lookbehind: true,
-			alias: 'comment'
-		}
-	});
-
-	Prism.languages.insertBefore('php', 'comment', {
+	Prism.languages.php = {
 		'delimiter': {
 			pattern: /\?>$|^<\?(?:php(?=\s)|=)?/i,
 			alias: 'important'
-		}
-	});
-
-	Prism.languages.insertBefore('php', 'keyword', {
+		},
+		'comment': [
+			{
+				pattern: /(^|[^\\])(?:\/\*[\s\S]*?\*\/|\/\/.*)/,
+				lookbehind: true
+			},
+			{
+				pattern: /(^|[^\\])#.*/,
+				alias: 'shell-comment',
```

Comment:
Is this alias necessary? If not, then we can combine both patterns into `/(^|[^\\])(?:\/\*[\s\S]*?\*\/|\/\/.*|#.*)/`.

Also, please make both `comment` patterns greedy.

Reply:
I removed `shell-comment` alias (it's not necessary).
But when pattern it is greedy, then `// ?>` is matched as comment (instead of `//` comment and `?>` delimiter)

Reply:
That's true. Then let's keep it non-greedy and hope for the best!